Balance levels, chunks, waves

Fenway17 opened the pull request:
Revamped Spider, Graveyard and Clown levels:
- Spider level now has 6 chunks and 4-7 waves per chunk
- Graveyard level now has 7 chunks and 4-7 waves per chunk
  - Fog introduced in the later chunks
  - Spider level enemies introduced in the later chunks
- Clown level now has 7 chunks and 4-7 waves per chunk
  - Fog introduced in the later chunks
  - Spider level enemies introduced in the later chunks

- Small fix to prevent tower health going below 0

- Major changes to towers to reduce stalling

TODO:
- Extensive testing on the levels

tanwayne890 commented:
Buff: 
- increase beyblade health a little, to last a bit longer
- spikey requires 75 ink and increase damage by 10, previously was 100 ink 

Nerf:
- laser tower requires 75 ink to build and each upgrade, previous was 50 to build and each upgrade

tanwayne890 commented:
General comments:
- high cost deck: usually hard at the start of level, as the game progress becomes easier
- heal doodle health is necessary since each chunk does not restore doodle health but restore base health.

Spider levels:
- for elemental spiders, using elemental towers still possible, since elemental spider does not immune 
to ice + water effects
- elemental spiders chunk usually ink overload (using high cost deck)
- as more towers has been build, build timer seems to be blocked 

Ghost level:
- mortar extremely good on ghost level, since the ghost walks slow
- mortar too op, might need reduced health and damage. increase elemental upgrade cost (?). done a little nerf https://github.com/CS3247-Game-Development-Team-6/Doodles/pull/379/commits/00cf7d7d8c3bdd6d437ab9308add006e619fad5b
- introducing fog is a good idea, but there is a bug where unable to clear even though ink is paid, so stucked here

![image](https://user-images.githubusercontent.com/58804560/200161197-dbe079c3-1c13-4d39-9b69-e5175d79077b.png)


Clown level:
- stuck at fog level too
- hardest level. strats: plant tower far away using mortar, barrierwall / landmine to make bomb drop faster, use slow (element effects / gel tower)
- **high priority bug** encountered on the tower fix cost exceeds ink max e.g. 400 tower fix cost, 300 max ink obtainable

Fenway17 committed:
Element balancing
- Fixed slow for ice effect
- Further reduced freeze stun effect

Fenway17 committed:
Tower balancing
- Increased health decay of gel tower
- Reduced enemy number in first wave of clown chunk 1

Fenway17 committed:
Tower balancing
- Reduced slow effect of gel tower

Fenway17 committed:
Tower balancing
- Reduced mortar range slightly
- Reduced mortar fire rate slightly
- Reduced gel slow effect duration
- Reduced ice chill slow effect duration

Fenway17 committed:
Tower balancing
- Further reduced slow and stun effects and duration (it is still too strong)

Fenway17 committed:
Levels balancing
- Reduced enemies in chunk 2 slightly
- Fixed range of gel tower

Fenway17 committed:
Tower balance
- increased gel tower range slightly

Fenway17 committed:
Clown level balancing
- Drastically reduced the number of surprise and curse clowns

Fenway17 committed:
Merge remote-tracking branch 'origin/Debug-GelTower-StopShootingWhen0HP' into balance-levels-chunks-waves

Fenway17 committed:
Merge remote-tracking branch 'origin/main-v2' into balance-levels-chunks-waves

# Conflicts:
#	Assets/ScriptableObjects/Levels/3_Clown_Chunk_1.asset
#	UserSettings/EditorUserSettings.asset
